### PR TITLE
Allow add/delete via addressbook update endpoint

### PR DESCRIPTION
**Pull request by nicholasyoder**
Fixes problem with editing a label that wasn't previously set. [Reported in discord.](https://discord.com/channels/657906918408585217/1080783409640648765/1288371098278428706)

### CODE REVIEW

File: rotkehlchen/tests/api/test_addressbook.py [hunk 1]
```diff
@@ -405,40 +405,57 @@ def test_update_addressbook(
...
+    with db_addressbook.read_ctx(book_type) as cursor:
+        entries_after_update = db_addressbook.get_addressbook_entries(cursor, filter_query=AddressbookFilterQuery.make())[0]  # noqa: E501
+    assert all(entry in entries_after_update for entry in new_entries)
+
+    # test that updating an entry with a blank label deletes the entry
+    entry_to_delete = AddressbookEntry(
+        address=ADDRESS_ETH,
+        name='',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    )
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_addressbook.py:445]: this one is not sent in the request below... how did it work? I suspect the assertion that you have below doesn't work as you expect

File: rotkehlchen/tests/api/test_addressbook.py [hunk 2]
```diff
@@ -405,40 +405,57 @@ def test_update_addressbook(
...
-        assert db_addressbook.get_addressbook_entries(cursor, filter_query=AddressbookFilterQuery.make())[0] == entries_in_db_before_bad_patch  # noqa: E501
+    response = requests.patch(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'entries': [entry.serialize() for entry in new_entries],
+        },
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_addressbook.py:433]:
```suggestion
        json={'entries': [entry.serialize() for entry in new_entries]},
```

File: rotkehlchen/db/addressbook.py [hunk 3]
```diff
@@ -128,9 +130,22 @@ def update_addressbook_entries(
...
                         continue  # skip update query, we just added a new entry instead
                     # InputError: single entry already exists, it must be updated using query below
+                elif (
+                    write_cursor.execute(
+                        'SELECT COUNT(*) FROM address_book WHERE address = ? AND blockchain = ?',
+                        (entry.address, entry_bockchain_value),
+                    ).fetchone()[0] == 0
+                ):
+                    # Add entry if it doesn't exist
+                    self.add_addressbook_entries(write_cursor, [entry])
```
[the remark this answers is not in the document]
Comment (nicholasyoder) [reply rotkehlchen/db/addressbook.py]: It has just checked here that this entry doesn't exist in the database, so if the insert fails, something strange is happening, and it should probably just raise the InputError without trying to handle it in any special way, so the user sees an error.

File: rotkehlchen/tests/api/test_addressbook.py (hunk 1 above)
Comment (nicholasyoder) [reply rotkehlchen/tests/api/test_addressbook.py]: I think it was working due to `entry_to_delete` having a blank name, so it didn't equal the event in the database. It should be updated to work correctly now.

File: rotkehlchen/tests/api/test_addressbook.py [hunk 4]
```diff
@@ -405,40 +405,55 @@ def test_update_addressbook(
...
+        ),
+        json={'entries': [entry.serialize() for entry in new_entries]},
+    )
+    assert_proper_sync_response_with_result(response)
+    with db_addressbook.read_ctx(book_type) as cursor:
+        entries_after_update = db_addressbook.get_addressbook_entries(cursor, filter_query=AddressbookFilterQuery.make())[0]  # noqa: E501
+    assert all(entry in entries_after_update for entry in new_entries)
+
+    # test that updating an entry with a blank label deletes the entry
+
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_addressbook.py:439]: This should be removed

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**